launch/launch/launch_description.py

```diff
@@ -45,12 +48,19 @@ class LaunchDescription(LaunchDescriptionEntity):
     def __init__(
         self,
         initial_entities: Optional[Iterable[LaunchDescriptionEntity]] = None,
+        *,
+        deprecated: Optional[Text] = None
     ) -> None:
         """Constructor."""
         self.__entities = list(initial_entities) if initial_entities is not None else []
+        self.__deprecated = deprecated
 
     def visit(self, context: LaunchContext) -> Optional[List[LaunchDescriptionEntity]]:
         """Override visit from LaunchDescriptionEntity to visit contained entities."""
+        if self.__deprecated is not None:
+            launch.logging.get_logger().warning(
+                'Deprecated launch file: {}'.format(self.__deprecated)
```

Comment:
It would be more useful if this included where or which launch description was deprecated. I'm try to remember if this information was already piped some where, as it could be captured as `<string>` (if done in an interpreter), or `some_launch_file_launch.py`, or `some_launch_file.launch.xml`, etc... It should be possible (technically) to ensure the launch description knows where it came from in most cases (when being included by another launch file or being run directly from `ros2 launch`, etc...).